src/vos/evt_iter.c

```diff
@@ -358,7 +387,16 @@ evt_iter_probe_sorted(struct evt_context *tcx, struct evt_iterator *iter,
 	index = evt_iter_probe_find(iter, rect);
 	if (index == -1)
 		return -DER_NONEXIST;
+
 	iter->it_index = index;
+	entry = evt_ent_array_get(iter->it_entries, index);
+	if (iter->it_forward)
+		entry->en_sel_ext.ex_lo = max(rect->rc_ex.ex_lo, entry->en_ext.ex_lo);
+	else
+		entry->en_sel_ext.ex_hi = min(rect->rc_ex.ex_hi, entry->en_ext.ex_hi);
```

Comment:
Same question, is it possible that "recx.ex_hi < anchor.ex_hi"?

Reply:
>Same question, is it possible that "recx.ex_hi < anchor.ex_hi"?

unless the anchor recx is deleted, otherwise it is impossible.

Reply:
Rather than fixing it here, it would be better, I think to fix the filter in evt_iter_probe_sorted.

Set the upper or lower bound of the range in the filter according to what is in the anchor.  That way, we don't have to do this here as it will already be set.   And if we are half way through the array, we can filter out and avoid sorting half of the rectangles.

So the variable you want to change is iter->it_filter